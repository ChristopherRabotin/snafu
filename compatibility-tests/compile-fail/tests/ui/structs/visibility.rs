// There are also happy-path tests

pub mod inner {
    use snafu::prelude::*;

    #[derive(Debug, Snafu)]
    pub(crate) struct Error;
}

fn private_is_default() {
    let _ = inner::Snafu.build();
}

fn main() {}
