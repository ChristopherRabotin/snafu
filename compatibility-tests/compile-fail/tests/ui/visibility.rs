// There are also happy-path tests

mod outer {
    pub mod inner {
        use snafu::prelude::*;

        #[derive(Debug, Snafu)]
        #[snafu(visibility(pub(crate)))]
        pub(crate) enum Error {
            PubCrate,
            #[snafu(visibility(pub(in crate::outer)))]
            PubInPath,
            #[snafu(visibility)]
            Private,
        }
    }

    fn private_is_applied() {
        let _ = self::inner::PrivateSnafu.build();
    }
}

fn pub_in_path_is_applied() {
    let _ = self::outer::inner::PubInPathSnafu.build();
}

fn private_is_applied() {
    let _ = self::outer::inner::PrivateSnafu.build();
}

fn main() {}
