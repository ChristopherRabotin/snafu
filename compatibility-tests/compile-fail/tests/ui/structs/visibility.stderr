error[E0603]: unit struct `Snafu` is private
  --> $DIR/visibility.rs:11:20
   |
11 |     let _ = inner::Snafu.build();
   |                    ^^^^^ private unit struct
   |
note: the unit struct `Snafu` is defined here
  --> $DIR/visibility.rs:6:21
   |
6  |     #[derive(Debug, Snafu)]
   |                     ^^^^^
   = note: this error originates in the derive macro `Snafu` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0624]: associated function `build` is private
  --> $DIR/visibility.rs:11:26
   |
11 |     let _ = inner::Snafu.build();
   |                          ^^^^^ private associated function
